src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.cs

```diff
@@ -3,22 +3,27 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
 using System.Diagnostics;
-using System.Dynamic.Utils;
 using System.Reflection;
 using System.Reflection.Emit;
 using System.Runtime.CompilerServices;
 
 namespace System.Linq.Expressions.Compiler
 {
+    internal interface ILocalCache
```

Comment:
Reusing locals in the `ILGen` helper requires passing the `LambdaCompiler` to it, so introducing this interface reduces the degree of coupling,